web/popminer/util.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+//go:build js && wasm
+
+package main
+
+import (
+	"runtime/debug"
+	"syscall/js"
+	"time"
+
+	"github.com/hemilabs/heminetwork/api"
+)
+
+var (
+	promiseConstructor = js.Global().Get("Promise")
+	objectConstructor  = js.Global().Get("Object")
+	arrayConstructor   = js.Global().Get("Array")
+)
+
+// Object represents a JavaScript Object.
+type Object map[string]any
```

Comment:
Cute. But you can directly return a map as well if that's what you are trying to achieve. I don;t think there is need for these gymnastics.

The point of having predefined return structures was that you could test them at the go level. Now this moves it up. I am ok with this but I am not seeing any advantage besides taste.

Reply:
Yes, a map could be used, however I did it like this so that it is both obvious that this will be represented in JS as an object (and as such the keys have meaning), and also easy to add utility functions for.

The dispatch handlers returning a `js.Value` also means that it is guaranteed that a `js.Value` can be returned when an error isn't.

Reply:
TODO: Use structs instead.